### DISCUSSION TITLE
Scaladoc play button questions

### DISCUSSION THREAD

**Discussion by ckipp01** [scala/scala3#15966] [General Discussion]
Having the play button in your docs is seemingly super useful. I see that it's being used on some of the projects that have public docs. You can see a couple examples here:

- https://virtuslab.github.io/scala-yaml/docs/quickstart.html
- https://mhammons.gitlab.io/slinc/

However, as expected pretty much all library documentation is going to want to import code from their codebase as examples. Currently, when you click play the payload to Scastie doesn't contain anything in the `libraries` section meaning it won't know anything about your library. So you assume you can run the documentation but you just get:

<img width="574" alt="Screenshot 2022-09-04 at 18 37 54" src="https://user-images.githubusercontent.com/13974112/188323947-15f289e5-7126-476c-9d47-c08650d1bce8.png">

What's the desired flow here? I imagine almost all libraries that will be using scaladocs as their doc site will have this sort of thing, will want to check that the snippet compiles (which it seems to when you create the docs), but then the play button will fail. I also then don't see a way to turn the play button off, unless you turn the compile checking off.

Is this just not something implemented? Are there plans for situations like this to work? Am I using/understanding this wrong?

**Comment by keynmol** [comment]
In Dotty sources I found an undocumented (?) option `-scastie-configuration` which I guess is intended to be passed to the Scastie embedding code, but I don't think it's hooked up anywhere in the code.

Is that right?

**Comment by SethTisue** [comment]
`-scastie-configuration` landed in https://github.com/lampepfl/dotty/pull/13258, by @pikinier20

there, it says:

> You can there specify libraryDependencies setting. It will probably be done better in future (automatically get this configuration from sbt) but for now it works like that.

as for Anton's question:

> I don't think it's hooked up anywhere in the code.

are you sure? it looks to me like `scaladoc/src/dotty/tools/scaladoc/renderers/Resources.scala` may access it via `ctx.args.scastieConfiguration`?

**Reply by julienrf** [reply]
I think the person who is currently in charge of Scaladoc at VirtusLab is now @szymon-rd (but AFAIK he can’t have a deep look right now)

**Reply by szymon-rd** [reply]
All of these issues will be addressed in the future. But that's true, it's an iterative process and we will allocate resources to next tasks related to scaladoc when we finish the current rework & fixes.

**Comment by SethTisue** [comment]
I can't find anything online where `scastieConfiguration` is set to anything but the empty string 🤷

**Reply by ckipp01** [reply]
Same, I created https://github.com/lampepfl/dotty/issues/16220.
